node/src/database/rocksdb.rs

```diff
@@ -7,130 +7,350 @@
 use super::{Registry, Tx, DB};
 use anyhow::Result;
 
+use dusk_consensus::commons::Block;
+use dusk_consensus::messages::Serializable;
+use rocksdb_lib::{
+    ColumnFamily, DBAccess, DBCommon, DBWithThreadMode, MultiThreaded,
+    OptimisticTransactionDB, OptimisticTransactionOptions, Options,
+    ReadOptions, SnapshotWithThreadMode, WriteOptions,
+};
+use std::{marker::PhantomData, path::Path, sync::Arc};
+
 enum TxType {
     ReadWrite,
     ReadOnly,
 }
 
-pub struct Backend {}
+const CF_LEDGER: &str = "cf_ledger";
+const CF_CANDIDATES: &str = "cf_candidates";
+
+pub struct Backend {
+    rocksdb: Arc<OptimisticTransactionDB>,
+}
 
 impl Backend {
-    fn begin_tx(&mut self, read_only: TxType) -> Transaction {
-        // TODO: rusk/issues/806 This should be addressed with another issue
-        // about integrating RocksDB
-        Transaction {}
+    fn begin_txn(
```

Comment:
Is `begin_tx`->`begin_txn` made on purpose?

Reply:
Updated.